sklearn/tests/test_impute.py

```diff
@@ -992,11 +999,40 @@ def test_missing_indicator_new(missing_values, arr_type, dtype, param_features,
     assert_allclose(X_trans_mask_sparse.toarray(), X_trans_mask)
 
 
-@pytest.mark.parametrize("param_sparse", [True, False, 'auto'])
-@pytest.mark.parametrize("missing_values", [np.nan, 0])
 @pytest.mark.parametrize(
     "arr_type",
-    [np.array, sparse.csc_matrix, sparse.csr_matrix, sparse.coo_matrix])
+    [sparse.csc_matrix, sparse.csr_matrix, sparse.coo_matrix,
+     sparse.lil_matrix, sparse.bsr_matrix])
+def test_missing_indicator_raise_on_sparse_with_missing_0(arr_type):
+    # test for sparse input and missing_value == 0
+
+    missing_values = 0
+    X_fit = np.array([[missing_values, missing_values, 1],
+                      [4, missing_values, 2]])
+    X_trans = np.array([[missing_values, missing_values, 1],
+                        [4, 12, 10]])
+
+    # convert the input to the right array format
+    X_fit = arr_type(X_fit)
+    X_trans = arr_type(X_trans)
+
+    indicator = MissingIndicator(missing_values=missing_values)
+
+    with pytest.raises(ValueError):
+        indicator.fit_transform(X_fit)
+    with pytest.raises(ValueError):
+        indicator.transform(X_trans)
+
```

Comment:
I'm not sure the second part raises the right error. I think the error raised is a non fitted estimator error since the fit should have failed above. To be sure, you can add a `match` parameter to `pytest.raises`, with a part of the error message you want to catch. 

You'll then need to fit it on proper data to check the transform.

Reply:
Good catch! As you said, it was raising "instance not fitted" exception instead of "sparse with missing_values=0". I fixed the test.